agent/consul/state/intention.go

```diff
@@ -333,6 +341,32 @@ func (s *Store) IntentionMatch(ws memdb.WatchSet, args *structs.IntentionQueryMa
 			}
 		}
 
+		// If matching an external uri, in addition to the exact match done
```

Comment:
This is the part of the change that I'm most worried about. I'm essentially re-implementing authorization logic here since I'm ensuring that trust domain intentions for `example.com` are matched for a uri `example.com/foo`.

I'm open to better ways to solve this.

Reply:
The logic here seems good to me.

Its possible that some other feature of memdb might make this easier by allowing prefix matching on the source. But that could be more risky as it would end up affecting other internal intentions. Also with compound indexing getting the prefix matching correct would be difficult and then hard to reason about in the future.

I think you did it the best way it could be done.